packages/core/src/Resource/Change/Relation/Relation.js

```diff
@@ -64,3 +64,26 @@ export function getChangedResourceRelations(resource) {
     ),
   ));
 }
+
+/**
+ * @param {HyralResource} resource
+ *
+ * @returns {HyralResource[]}
+ */
+export function getDeletedOneToOneRelatedResources(resource) {
+  if (!resource.relationships) {
+    return [];
+  }
+
+  if (resourceIsNew(resource) || previousState(resource.stateStack) === null) {
+    return [];
+  }
+
+  return Object.keys(resource.relationships)
+    .filter(relation => resource.relationships[relation].cardinality === 'one-to-one')
```

Comment:
doesn't it need to be recursive?

Reply:
you mean for every related resource as well? In that case: no, this will already be handled in the TaskGenerator, here only relations for the current resource should be handled.